### PR TITLE
Extract mustard checker into its own lib

### PR DESCRIPTION
**Pull request by matthew-andrews**

### CODE REVIEW

File: views/layout.html [hunk 1]
```diff
@@ -6,15 +6,7 @@
 		<meta name="dfp_site" content="test.next.ft.com" />
 		<link rel="stylesheet" href="/{{__name}}/main.css"/>
 		<title>Financial Times</title>
-		{{#if @root.flags.javascript.isSwitchedOn}}
-		<script>
-			var cutsTheMustard = 'matches' in Element.prototype;
```
Comment (aintgoin2goa) [inline views/layout.html]: Look at this [http://caniuse.com/#feat=matchesselector] it looks like this check will exclude all IEs??

Comment (matthew-andrews) [inline views/layout.html]: Yup.

**Comment by adgad** [comment]
I think :+1: to this, and move the matches discussion into next-mustard

File: views/layout.html (hunk 1 above)
Comment (matthew-andrews) [inline views/layout.html]: Until we have a polyfill our code breaks in IE.  This mustard check just reflects that reality — not actually changing anything.

**Comment by matthew-andrews** [comment]
Discussion moved here: https://github.com/Financial-Times/next-mustard/issues/1

**Pull request merged**